### DISCUSSION TITLE
Implementing oAuth flows with the Capacitor Browser

### DISCUSSION THREAD

**Discussion by leo-petrucci** [ionic-team/capacitor#4979] [Q&A]
Has anyone successfully managed to get an oAuth flow working with Capacitor?

So far I've got:
- Get Login url
- Open native browser to the url
- Let user finish logging in and be redirected

I was then thinking of having a specific landing page API side which would listen to an event once the callback was complete, that would then close the browser, but [according to the docs close() is only available on web and iOS???](https://capacitorjs.com/docs/apis/browser#close)

What does this mean? Can I not close the in-app browser from my app code? Any workarounds?

**Comment by Kikketer** [comment]
I've gotten close with Capacitor and Supabase: https://cjweed.com/capacitor-supabase-social-auth-react/

But I still haven't figured out how to tell the app to open the browser in the in-app browser. This flow uses the currently installed browser which can cause some issues:

1. You have to use the "consent" flow, redirects don't open the native app if you don't use that flow (annoying)
2. Other browsers may not even work after this as well, Brave and Orion do not work... Safari does.

**Reply by leo-petrucci** [reply]
I think I ended up figuring out that you can use deep links to automatically close the in-app browser. 

I don't have an example, nor have I worked with Capacitor in years, but I'm pretty sure that if you set your callback URL to be a deep-link opened by your app it'll close the in-app browser on its own.
